examples/widgets/check_buttons.py

```diff
@@ -25,21 +25,23 @@
 l2, = ax.plot(t, s2, lw=2, color='g', label='6 Hz')
 fig.subplots_adjust(left=0.2)
 
-lines = [l0, l1, l2]
+lines = {l.get_label(): l for l in [l0, l1, l2]}
 
 # Make checkbuttons with all plotted lines with correct visibility
 rax = fig.add_axes([0.05, 0.4, 0.1, 0.15])
-labels = [str(line.get_label()) for line in lines]
-visibility = [line.get_visible() for line in lines]
-check = CheckButtons(rax, labels, visibility)
+check = CheckButtons(
+    ax=rax,
+    labels=list(lines),
+    actives=[l.get_visible() for l in lines.values()]
+)
 
 
-def func(label):
-    index = labels.index(label)
-    lines[index].set_visible(not lines[index].get_visible())
+def callback(label):
+    ln = lines[label]
+    ln.set_visible(not ln.get_visible())
```

Comment:
```suggestion
    line = next(l for l in lines if l.get_label() == label)
    line.set_visible(not line.get_visible())
```

Reply:
I don't think using `next` is any less confusing than a dictionary, perhaps even more so. The alternative is using a full `for` loop, and at that point a dictionary lookup just seems less work.